depin/services/roles/ethereum/tasks/commands/install.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: Install dependencies
+  become: true
+  ansible.builtin.apt:
+    name:
+      - openjdk-21-jdk
+    state: present
+    update_cache: true
+
+- name: Create program folder for besu
+  become: true
+  ansible.builtin.file:
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+    dest: /opt/ethereum_besu/{{ ethereum_besu_version }}
+
+- name: Create program folder for teku
+  become: true
+  ansible.builtin.file:
+    state: directory
+    owner: root
+    group: root
+    mode: '0755'
+    dest: /opt/ethereum_teku/{{ ethereum_teku_version }}
+
+- name: Download besu binary  to tmp
+  become: true
+  ansible.builtin.unarchive:
+    remote_src: true
+    src: "{{ ethereum_besu_download_url }}"
+    dest: /tmp
+  failed_when: false
+
+- name: Download teku binary to tmp
+  become: true
+  ansible.builtin.unarchive:
+    remote_src: true
+    src: "{{ ethereum_teku_download_url }}"
+    dest: /tmp
+  failed_when: false
+
+- name: Find besu binary
+  become: true
+  ansible.builtin.find:
+    paths:
+      - /tmp
+      - "{{ ethereum_besu_download_path | default('/tmp') }}"
+    patterns: 'besu$'
+    use_regex: true
+    get_checksum: true
+    recurse: true
+  register: _download_besu
+
+- name: Find teku binary
+  become: true
+  ansible.builtin.find:
+    paths:
+      - /tmp
+      - "{{ ethereum_teku_download_path | default('/tmp') }}"
+    patterns: 'teku$'
+    use_regex: true
+    get_checksum: true
+    recurse: true
+  register: _download_teku
```

Comment:
Should try and leverage `depin.core.installer` here.. but might need to run it twice. Once for the execution and once for the consensus clients to make it less "spaghetti"

Reply:
structure of files and folders the same like in Nerdnode gitbook